src/initial_conditions.jl

```diff
@@ -68,6 +71,24 @@ function initialize_from_rest(M::PrimitiveEquationModel)
     return progn
 end
 
+function initialize_from_vector!(progn_new::PrognosticVariables{NF}, ic::Vector{T}, M::ModelSetup) where {NF, T<:LowerTriangularMatrix}
+   
+    @unpack nlev = M.geometry
+    @assert length(ic) == nlev
+
+    # SPECTRAL TRUNCATION/INTERPOLATION to new resolution and conversion to NF
+    for (layer_new, layer_ic) in zip(progn_new.layers, ic)
+        layer_new_lf1 = layer_new.leapfrog[1]
+        vor_new = layer_new_lf1.vor
+        lmax,mmax = size(vor_new) .- (2,1)
+
+        vor_ic_trunc = spectral_truncation(layer_ic, lmax+1, mmax)
+        vor_new .=  convert(LowerTriangularMatrix{Complex{NF}}, vor_ic_trunc)
```

Comment:
Note that we still have #135 open, meaning that `vor_new .= some_array` may not do what you expect it too. To avoid this broadcasting issue in the first place I've already defined `copyto!` for `:LowerTriangularMatrix` which can handle the type conversion in `convert` implicitly.

Reply:
This is just copied from the `initialize_from_file!` function. Then that should also be looked at, but we can also replace that function with `set_vorticity!` etc as well

Reply:
Yes, good point!